### PR TITLE
partio 1.17.0: add missing googletest dependency

### PR DESCRIPTION
**Pull request by davvid**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

There is 1 pre-existing issue where the modules link to Python.framework but I'm not sure how to fix that (cmake's FindPythonLibs seems to do that by default).

This fixes being able to build the latest version and is no worse than before.

### CODE REVIEW

File: Formula/partio.rb [hunk 1]
```diff
@@ -1,23 +1,13 @@
...
-    rebuild 1
-    sha256 cellar: :any_skip_relocation, arm64_ventura:  "f23723988ee0c4fd13e2ef3999cd48cdc0713e2e68b08f1e1e8fb64bfb72735e"
-    sha256 cellar: :any_skip_relocation, arm64_monterey: "e8779e2d241646c96f0b6fa412219f0556b69ff03d9dd98d6a9ecb1172baccf9"
-    sha256 cellar: :any_skip_relocation, arm64_big_sur:  "40c077714c5b73b2d1eca4b57054262ca3e4bef68ec9827ab39f205a3d7d170b"
-    sha256 cellar: :any_skip_relocation, ventura:        "39707a6cc21e84edc6cae4c82637600e049b521e92fe239dcc545b9c8bbc79dd"
-    sha256 cellar: :any_skip_relocation, monterey:       "247dad3b9b36c9a485ad9142e3840e0dd503547c8faf0b6a3ec6f92195278b01"
-    sha256 cellar: :any_skip_relocation, big_sur:        "4fa3eb20d12124955b39eb260a96586bedd4f254d5bbae6f52a0cdf186d380ea"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "ff7788987e0e5e2dcfc950be3b9f229f7061c2b6f0a6f6decda35bdf909497d1"
-  end
-
```
Comment (Moisan) [inline Formula/partio.rb:18]: Please do not change the bottle block. It will be automatically updated when merging.

**Comment by carlocab** [comment]
partio 1.17.0 was already shipped in #135576.

File: Formula/partio.rb [hunk 2]
```diff
@@ -29,6 +19,8 @@ class Partio < Formula
   def install
     args = std_cmake_args
     args << "-DPARTIO_USE_GLVND=OFF" unless OS.mac?
+    args << "-DPARTIO_GTEST_ENABLED=ON"
+    args << "-DGTEST_LOCATION=#{Formula["googletest"].opt_prefix}"
```
Comment (carlocab) [inline Formula/partio.rb:23]: Why is this needed?

Comment (davvid) [reply Formula/partio.rb:33]: The build scripts (currently ~ in v1.17.0 and older) fail to build when using an alternate (non-`/usr/local`) brew location without specifying the googletest location.

There's already a fix[1] merged upstream that makes googletest truly optional (rather than the build error that we'd hit in 1.17.0 and older) so it's not too big deal since v1.17.1 won't try to use googletest unless it's available.

I updated the commit message to mention these details, but I also closed this MR since v1.17.1 will make it obsolete.

[1] https://github.com/wdas/partio/commit/2200a2349be560ece792d4f174a8daa8af76c33a
